### DISCUSSION TITLE
https://ericswpark.com/blog/2022/2022-11-07-smooth-scrubbing-videos/

### DISCUSSION THREAD

**Discussion by giscus** [ericswpark/ericswpark.github.io-comments#11] [Announcements]
# https://ericswpark.com/blog/2022/2022-11-07-smooth-scrubbing-videos/

Student Developer

https://ericswpark.com/blog/2022/2022-11-07-smooth-scrubbing-videos/

**Comment by porg** [comment]
My inquiry was:

- I want to get a small video file size (=benefit from inter-frame encoding),
- while still achieve the best possible scrubbing performance.
- Which inter-frame settings shall I use for encoding to achieve that goal?

You article answers that general question quite well! Thanks!

Ideally I would love it to get a bit extended by this:
- How is HEVC handling GOP and I-frames, P-frames, B-frames?
  - How can I control these parameters with ffmpeg or Handbrake?
  - How can I analyze videos with ffprobe and mediainfo in that regard (as of 2024-10 both tools provide great info for h264 in that regard, but none for HEVC)
- What about the container formats? Can their setup and metadata also aid in scrubbing performance?
  - I assume it mostly depends from the codec itself, but could imagine that the container in its manifest can also provide some data aiding efficient scrubbing?

**Reply by ericswpark** [reply]
For HEVC, you'd have to look at the encoder preferences to control these parameters. I'd imagine the team behind ffmpeg/ffprobe will add more capabilities regarding HEVC in the future, although maybe not depending on which codec replaces H264 in terms of popularity.

I don't believe different container formats would impact scrubbing performance, but perhaps it's an interesting experiment for another blog post. Any modern popular container format like MP4/MKV should be equivalent in performance if my understanding is correct and as long as the implementation to open these containers on different players/devices don't have any performance-degrading bugs.
